skbio/stats/ordination/tests/test_canonical_correspondence_analysis.py

```diff
@@ -0,0 +1,174 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+
+import numpy as np
+import numpy.testing as npt
+import pandas as pd
+from unittest import TestCase, main
+
+from skbio import OrdinationResults
+from skbio.stats.ordination import cca
+from skbio.util import get_data_path, assert_ordination_results_equal
+
+
+class TestCCAErrors(TestCase):
+    def setUp(self):
+        """Data from table 11.3 in Legendre & Legendre 1998."""
+        self.Y = pd.DataFrame(np.loadtxt(get_data_path('example3_Y')))
+        self.X = pd.DataFrame(np.loadtxt(get_data_path('example3_X')))
+
+    def test_shape(self):
+        X, Y = self.X, self.Y
+        with npt.assert_raises(ValueError):
+            cca(Y, X[:-1])
+
+    def test_Y_values(self):
+        X, Y = self.X, self.Y
+        Y[0, 0] = -1
+        with npt.assert_raises(ValueError):
+            cca(Y, X)
+        Y[0] = 0
+        with npt.assert_raises(ValueError):
+            cca(Y, X)
+
+    def test_scaling(self):
+        X, Y = self.X, self.Y
+        with npt.assert_raises(NotImplementedError):
+            cca(Y, X, 3)
+
+    def test_all_zero_row(self):
+        X, Y = pd.DataFrame(np.zeros((3, 3))), pd.DataFrame(np.zeros((3, 3)))
+        with npt.assert_raises(ValueError):
+            cca(X, Y)
+
+
+class TestCCAResults1(TestCase):
+    def setUp(self):
+        """Data from table 11.3 in Legendre & Legendre 1998
+        (p. 590). Loaded results as computed with vegan 2.0-8 and
+        compared with table 11.5 if also there."""
+        self.feature_ids = ['Feature0', 'Feature1', 'Feature2', 'Feature3',
+                            'Feature4', 'Feature5', 'Feature6', 'Feature7',
+                            'Feature8']
+        self.sample_ids = ['Sample0', 'Sample1', 'Sample2', 'Sample3',
+                           'Sample4', 'Sample5', 'Sample6', 'Sample7',
+                           'Sample8', 'Sample9']
+        self.env_ids = ['Constraint0', 'Constraint1',
+                        'Constraint2', 'Constraint3']
+        self.pc_ids = ['CCA1', 'CCA2', 'CCA3', 'CCA4', 'CCA5', 'CCA6', 'CCA7',
+                       'CCA8', 'CCA9']
+        self.Y = pd.DataFrame(
+            np.loadtxt(get_data_path('example3_Y')),
+            columns=self.feature_ids,
+            index=self.sample_ids)
+        self.X = pd.DataFrame(
+            np.loadtxt(get_data_path('example3_X')),
+            columns=self.env_ids,
+            index=self.sample_ids
+            ).ix[:, :-1]
+
```

Comment:
Just removed it, let's see if the tests pass :O

On (Jul-13-15|16:26), Jorge Cañardo Alastuey wrote:
> > +                           'Sample4', 'Sample5', 'Sample6', 'Sample7',
> > +                           'Sample8', 'Sample9']
> > +        self.env_ids = ['Constraint0', 'Constraint1',
> > +                        'Constraint2', 'Constraint3']
> > +        self.pc_ids = ['CCA1', 'CCA2', 'CCA3', 'CCA4', 'CCA5', 'CCA6', 'CCA7',
> > +                       'CCA8', 'CCA9']
> > +        self.Y = pd.DataFrame(
> > +            np.loadtxt(get_data_path('example3_Y')),
> > +            columns=self.feature_ids,
> > +            index=self.sample_ids)
> > +        self.X = pd.DataFrame(
> > +            np.loadtxt(get_data_path('example3_X')),
> > +            columns=self.env_ids,
> > +            index=self.sample_ids
> > +            ).ix[:, :-1]
> > +
> 
> I believe it can be dropped.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/scikit-bio/pull/925/files#r34523618